flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/failover/flip1/RestartBackoffTimeStrategyFactoryLoaderTest.java

```diff
@@ -0,0 +1,210 @@
+	}
+
+	@Test
+	public void testNoStrategySpecifiedWhenCheckpointingDisabled() throws Exception {
+		final RestartBackoffTimeStrategy.Factory factory =
+			RestartBackoffTimeStrategyFactoryLoader.createRestartStrategyFactory(
+				DEFAULT_RESTART_STRATEGY_CONFIGURATION,
+				new Configuration(),
+				false);
+
+		assertThat(
+			factory,
+			instanceOf(NoRestartBackoffTimeStrategy.NoRestartBackoffTimeStrategyFactory.class));
+	}
+
+	@Test
+	public void testLegacyStrategySpecifiedInJobConfig() throws Exception {
+		// the fixed delay strategy config in cluster config should not take effect
+		// as it will be overridden by the failure rate strategy config in job config
+		final Configuration conf = new Configuration();
+		conf.setString(ConfigConstants.RESTART_STRATEGY, "fixed-delay");
+
+		final RestartBackoffTimeStrategy.Factory factory =
+			RestartBackoffTimeStrategyFactoryLoader.createRestartStrategyFactory(
+				new RestartStrategies.FailureRateRestartStrategyConfiguration(
+					1,
+					Time.milliseconds(1000),
+					Time.milliseconds(1000)),
+				conf,
+				false);
+
+		assertThat(
+			factory,
+			instanceOf(FailureRateRestartBackoffTimeStrategy.FailureRateRestartBackoffTimeStrategyFactory.class));
+	}
+
+	@Test
+	public void testLegacyStrategySpecifiedInClusterConfig() throws Exception {
+		// the fixed delay strategy config in cluster config should not take effect
+		// as it will be overridden by the failure rate strategy config in job config
+		final Configuration conf = new Configuration();
+		conf.setString(ConfigConstants.RESTART_STRATEGY, "fixed-delay");
+
+		final RestartBackoffTimeStrategy.Factory factory =
+			RestartBackoffTimeStrategyFactoryLoader.createRestartStrategyFactory(
+				DEFAULT_RESTART_STRATEGY_CONFIGURATION,
+				conf,
+				false);
+
+		assertThat(
+			factory,
+			instanceOf(FixedDelayRestartBackoffTimeStrategy.FixedDelayRestartBackoffTimeStrategyFactory.class));
+	}
+
+	@Test
+	public void testDeprecatedValuesSpecifiedInClusterConfig() throws Exception {
+		// the fixed delay strategy config in cluster config should not take effect
+		// as it will be overridden by the failure rate strategy config in job config
+		final Configuration conf = new Configuration();
+		conf.setInteger(ConfigConstants.RESTART_STRATEGY_FIXED_DELAY_ATTEMPTS, 1);
+
+		final RestartBackoffTimeStrategy.Factory factory =
+			RestartBackoffTimeStrategyFactoryLoader.createRestartStrategyFactory(
+				DEFAULT_RESTART_STRATEGY_CONFIGURATION,
+				conf,
+				false);
+
+		assertThat(
+			factory,
+			instanceOf(FixedDelayRestartBackoffTimeStrategy.FixedDelayRestartBackoffTimeStrategyFactory.class));
+	}
+
+	// ------------------------------------------------------------------------
+	//  utilities
+	// ------------------------------------------------------------------------
+
+	/**
+	 * A test implementation of {@link RestartBackoffTimeStrategy}.
+	 */
+	private static class TestRestartBackoffTimeStrategy implements RestartBackoffTimeStrategy {
```

Comment:
Seems a duplicated class of `TestRestartBackoffTimeStrategy` with that in TestRestartBackoffTimeStrategy.java

Reply:
You are right. Will modify the standalone TestRestartBackoffTimeStrategy class and use it for testing.